lib/mail/network/retriever_methods/imap.rb

```diff
@@ -153,8 +153,13 @@ def validate_options(options)
       # Start an IMAP session and ensures that it will be closed in any case.
       def start(config=Mail::Configuration.instance, &block)
         raise ArgumentError.new("Mail::Retrievable#imap_start takes a block") unless block_given?
-
-        imap = Net::IMAP.new(settings[:address], settings[:port], settings[:enable_ssl], nil, false)
+        tmp_ssl = settings[:enable_ssl]
+        tmp_ssl = false unless settings[:starttls_options].nil?
+        
+        imap = Net::IMAP.new(settings[:address], settings[:port], tmp_ssl, nil, false)
+        if settings[:starttls_options].is_a?(Hash)
```

Comment:
Providing an options hash to enable/disable the feature is a bit awkward, particular if you don't need to set any options. Would expose this as a boolean: `starttls: true`